### ISSUE TITLE
Systems only support 1 compatible component

### ISSUE THREAD

**Issue by anthony-y** [anthony-y/sharp-ecs#1]

**Comment by deccer** [comment]
so we remove the exception, and allow multiple components per entity.
and use something like <code>entity.GetComponent&lt;TransformComponent&gt;("leftThrusterTF")</code>. Yes or No?

**Comment by anthony-y** [comment]
That's actually a good idea but it's not what this issue is about :)

It's about [Entity Systems](https://github.com/anthony-y/sharp-ecs/tree/master/SharpECS/EntitySystem.cs) and how they can only have 1 component type that they look for on entities and since C# doesn't let you have a variable amount of type parameters, the only way I think of doing it is something like:

```
class EntitySystem 
{
    private List<Type> _compatibleComponents;

    public EntitySystem(params Type[] compatibleComponentTypes)
    {
        foreach (var c in compatibleComponentTypes)
        {
            if (!(c is IComponent)) { throw new Exception("Type passed was not a child type IComponent"); }
            else { _compatibleComponents.Add(c); }
        }
    }
}
```

**Comment by anthony-y** [comment]
But I will get to work implementing that `entity.GetComponent<TransformComponent>("leftThrusterTF")` tomorrow :)

**Comment by anthony-y** [comment]
It's in :)

But it's really awkward to implement in systems :/

**Comment by deccer** [comment]
Hmm, maybe the systems themselves could be different, there might be stupid systems who don't care about multiple components per type, and others who take the "name" or "tag" or "id" into consideration. Unless its total nonsense anyway

**Comment by anthony-y** [comment]
Also I don't know if you saw but I started working on this issue (Systems only support 1 compatible component) in branch ["multiple-compatible"](https://github.com/anthony-y/sharp-ecs/tree/multiple-compatible). The code compiles and runs but no entities are being drawn and as far as I know, updated.

**Comment by deccer** [comment]
oh. I have not seen that yet. I will check it out - literally hehe

**Comment by anthony-y** [comment]
:laughing:  took me a second to get that one :)

**Comment by deccer** [comment]
Player gets drawn now. Enemy/Enemies not yet. #3

**Comment by anthony-y** [comment]
Implemented in 1b02ada :smile:
